src/vtt-segment-loader.js

```diff
@@ -506,34 +476,30 @@ export default class SegmentLoader extends videojs.EventTarget {
       return;
     }
 
-    let isEndOfStream = detectEndOfStream(this.playlist_,
-                                          this.mediaSource_,
-                                          segmentInfo.mediaIndex);
+    // TODO is this check important
+    // if (segmentInfo.mediaIndex === this.playlist_.segments.length - 1 &&
+    //     this.mediaSource_.readyState === 'ended' &&
+    //     !this.seeking_()) {
+    //   return;
+    // }
 
-    if (isEndOfStream) {
-      this.mediaSource_.endOfStream();
-      return;
-    }
+    if (this.syncController_.timestampOffsetForTimeline(segmentInfo.timeline) === null) {
+      // We don't have the timestamp offset that we need to sync subtitles.
+      // Rerun on a timestamp offset or user interaction.
+      let checkTimestampOffset = () => {
+        this.syncController_.off('timestampoffset', checkTimestampOffset);
+        this.state = 'READY';
+        if (!this.paused()) {
+          // if not paused, queue a buffer check as soon as possible
+          this.monitorBuffer_();
+        }
+      };
 
-    if (segmentInfo.mediaIndex === this.playlist_.segments.length - 1 &&
-        this.mediaSource_.readyState === 'ended' &&
-        !this.seeking_()) {
+      this.syncController_.on('timestampoffset', checkTimestampOffset);
+      this.state = 'WAITING_ON_TIMELINE';
```

Comment:
While the state is not `WAITING` any aborts will just clear pending segment (which at this point there isn't one yet) and does nothing else. The callback will set the state to ready and call `monitorBuffer_` to restart the lifecycle. Since it starts back from the beginning, any seeks or subtitle changes that happened in the waiting time shouldn't have adverse effects